lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -553,21 +631,19 @@ def write(chunk)
           end
         else
           begin
-            # Does the actual write to the cloud logging api.
-
-            client = api_client
-
-            # The URI of the write is constructed by the Google::Api request;
-            # it is equivalent to this URL:
-            # 'https://logging.googleapis.com/v1beta3/projects/' \
-            #   "#{@project_id}/logs/#{log_name}/entries:write"
             write_request = \
-              Google::Apis::LoggingV1beta3::WriteLogEntriesRequest.new(
-                common_labels: labels,
+              Google::Apis::LoggingV2beta1::WriteLogEntriesRequest.new(
+                log_name: log_name,
+                resource: Google::Apis::LoggingV2beta1::MonitoredResource.new(
+                  type: group_resource_type,
+                  labels: group_resource_labels
+                ),
+                labels: group_common_labels,
                 entries: entries)
+            write_request.labels = nil if write_request.labels.empty?
```

Comment:
I traced back to the original commit that added this logic and copied the comment from there:
"# Remove the labels if we didn't populate it with anything.".